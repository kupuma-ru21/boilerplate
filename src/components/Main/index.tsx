const Main = () => {
  return (
    <main>
      <h1>React Advance</h1>
    </main>
  );
};

export default Main;
