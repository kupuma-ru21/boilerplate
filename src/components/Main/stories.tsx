import { withKnobs, text } from '@storybook/addon-knobs';
import Main from '.';

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs],
};

export const Basic = () => {
  return (
    <Main
      title={text('Title', 'React Advance')}
      description={text(
        'Description',
        'TypeScript, ReactJS, NextJS e Styled Components'
      )}
    />
  );
};
