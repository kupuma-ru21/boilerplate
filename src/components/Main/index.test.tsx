import { render, screen } from '@testing-library/react';
import Main from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />);
    screen.getByRole('heading', { name: 'React Advance' });

    expect(container.firstChild).toMatchSnapshot();
  });
});
