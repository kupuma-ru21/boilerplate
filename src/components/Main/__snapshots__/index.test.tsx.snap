// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main>
  <h1>
    React Advance
  </h1>
</main>
`;
